diff --git a/cc/config/x86_darwin_host.go b/cc/config/x86_darwin_host.go
index bb751feb..ff2b3f0d 100644
--- a/cc/config/x86_darwin_host.go
+++ b/cc/config/x86_darwin_host.go
@@ -89,6 +89,7 @@ var (
 		"10.10",
 		"10.11",
 		"10.12",
+		"10.13",
 	}
 
 	darwinAvailableLibraries = append(
diff --git a/ui/build/java.go b/ui/build/java.go
index 473af012..a9f9129e 100644
--- a/ui/build/java.go
+++ b/ui/build/java.go
@@ -16,7 +16,6 @@ package build
 
 import (
 	"regexp"
-	"runtime"
 	"strings"
 	"sync"
 )
@@ -104,33 +103,19 @@ func checkJavaVersion(ctx Context, config Config) {
 		ctx.Fatalln("stop")
 	}
 
-	if runtime.GOOS == "linux" {
-		// Early access builds of OpenJDK 9 do not contain the string "openjdk" in the
-		// version name. TODO(tobiast): Reconsider once the OpenJDK 9 toolchain is stable.
-		// http://b/62123342
-		if !strings.Contains(java_version, "openjdk") && !experimental_use_openjdk9 {
-			ctx.Println("*******************************************************")
-			ctx.Println("You are attempting to build with an unsupported JDK.")
-			ctx.Println()
-			ctx.Println("Only an OpenJDK based JDK is supported.")
-			ctx.Println()
-			ctx.Println("Please follow the machine setup instructions at:")
-			ctx.Println("    https://source.android.com/source/initializing.html")
-			ctx.Println("*******************************************************")
-			ctx.Fatalln("stop")
-		}
-	} else { // darwin
-		if strings.Contains(java_version, "openjdk") {
-			ctx.Println("*******************************************************")
-			ctx.Println("You are attempting to build with an unsupported JDK.")
-			ctx.Println()
-			ctx.Println("You use OpenJDK, but only Sun/Oracle JDK is supported.")
-			ctx.Println()
-			ctx.Println("Please follow the machine setup instructions at:")
-			ctx.Println("    https://source.android.com/source/initializing.html")
-			ctx.Println("*******************************************************")
-			ctx.Fatalln("stop")
-		}
+	// Early access builds of OpenJDK 9 do not contain the string "openjdk" in the
+	// version name. TODO(tobiast): Reconsider once the OpenJDK 9 toolchain is stable.
+	// http://b/62123342
+	if !strings.Contains(java_version, "openjdk") && !experimental_use_openjdk9 {
+		ctx.Println("*******************************************************")
+		ctx.Println("You are attempting to build with an unsupported JDK.")
+		ctx.Println()
+		ctx.Println("Only an OpenJDK based JDK is supported.")
+		ctx.Println()
+		ctx.Println("Please follow the machine setup instructions at:")
+		ctx.Println("    https://source.android.com/source/initializing.html")
+		ctx.Println("*******************************************************")
+		ctx.Fatalln("stop")
 	}
 
 	incompatible_javac := strings.Contains(javac_version, incompatibleJavacStr)
